### PR TITLE
Moved embedded css into .less file. Removed a lot of custom code by u…

### PR DESCRIPTION
**Pull request by acolebourne**
…sing vanilla dataTables. Improved logic of the A-Z index and Category sort buttons. Made the 'entry' page the default link.

### CODE REVIEW

File: uportal-war/src/main/webapp/media/skins/respondr/defaultSkin/skin.xml [hunk 1]
```diff
@@ -42,6 +42,7 @@
   <css included="plain" resource="true">/scripts/jstree/style.css</css>
   <css included="aggregated" resource="true">/scripts/jstree/style.min.css</css>
 
+  <css import="true">../common/common_skin.xml</css>
```
Comment (drewwills) [inline uportal-war/src/main/webapp/media/skins/respondr/defaultSkin/skin.xml]: Looks like library CSS for...

* datatables/1.9.4/plugins/integration/bootstrap/3;  and
* datatables/1.9.4/extras/column-filter-widgets

wasn't being included at all.

File: uportal-war/src/main/webapp/media/skins/respondr/common/common_skin.xml [hunk 2]
```diff
@@ -23,6 +23,14 @@
...
  xsi:schemaLocation="http://www.jasig.org/uportal/web/skin https://source.jasig.org/schemas/resource-server/skin-configuration/skin-configuration-v1.2.xsd">
 
+	<!-- Data Table Bootstrap Includes -->
+    <css included="plain" resource="true">/rs/datatables/1.9.4/plugins/integration/bootstrap/3/dataTables.bootstrap.css</css>
+    <css included="aggregated" resource="true">/rs/datatables/1.9.4/plugins/integration/bootstrap/3/dataTables.bootstrap.css</css>
+    
+	<!-- Data Table ColumnFilterWidgets Includes -->    <!-- Data Table ColumnFilterWidgets Includes -->
+	<css included="plain" resource="true">/rs/datatables/1.9.4/extras/column-filter-widgets/css/ColumnFilterWidgets.css</css>
+    <css included="aggregated" resource="true">/rs/datatables/1.9.4/extras/column-filter-widgets/css/ColumnFilterWidgets.css</css>
+
```
Comment (drewwills) [inline uportal-war/src/main/webapp/media/skins/respondr/common/common_skin.xml]: Do we really want to reorder these statements?  Or was this change an early attempt to remedy some missing CSS?

Comment (acolebourne) [inline uportal-war/src/main/webapp/media/skins/respondr/common/common_skin.xml]: I think the schema specifies <css> followed by <js>

Comment (MichaelVose2) [inline uportal-war/src/main/webapp/media/skins/respondr/common/common_skin.xml]: Anthony is correct:

https://source.jasig.org/schemas/resource-server/skin-configuration/skin-configuration-v1.2.xsd

<xs:element name="resources"><xs:complexType><xs:sequence><xs:element name="parameter" type="parameter" minOccurs="0" maxOccurs="unbounded"/><xs:element name="css" type="css" minOccurs="0" maxOccurs="unbounded"/><xs:element name="js" type="js" minOccurs="0" maxOccurs="unbounded"/></xs:sequence></xs:complexType></xs:element>

File: uportal-war/src/main/webapp/WEB-INF/jsp/Marketplace/portlet/entry.jsp [hunk 3]
```diff
@@ -462,42 +239,52 @@
...
-                                        <h5><c:out value="${relatedPortlet.title}"/></h5>
-                                        <p><c:out value="${relatedPortlet.description}"/></p>
-                                    </div>
-                                </a>
-                            </div>
-                        </div>
-                    </div>
-                    <c:if test="${(status.index + 1) mod 4 == 0}">
+                    
+                    <div class = "image-panel-container">
```
Comment (MichaelVose2) [inline uportal-war/src/main/webapp/WEB-INF/jsp/Marketplace/portlet/entry.jsp]: For this to look acceptable images may need to be created for each portlet.

File: uportal-war/src/main/webapp/WEB-INF/jsp/Marketplace/portlet/entry.jsp [hunk 4]
```diff
@@ -387,19 +129,78 @@
...
+					<button type="button" class="btn btn-default btn-sm dropdown-toggle marketplace_dropdown_button" data-toggle="dropdown">
+						<spring:message code="options" text="Options"/>
+						<span class="caret"></span>
+					</button>
+					<ul class="dropdown-menu marketplace_dropdown_menu" role="menu">
+					</ul>
+					<button type="button" class="btn btn-default btn-sm" data-toggle="modal" data-target="#${n}-feedback-modal"><span class = "fa fa-pencil"></span><span class = "padding-left">Rate this application</span></button>
+				</div>
+  				<!-- Modal -->
+				<div class="modal fade" id="${n}-feedback-modal" role="dialog">
```
Comment (MichaelVose2) [inline uportal-war/src/main/webapp/WEB-INF/jsp/Marketplace/portlet/entry.jsp]: Review this portlet form now appears in a dialog, as opposed to displaying in the page on load.

**Comment by acolebourne** [comment]
Some improvements have come to light, closing pull request to be re-presented at a later date.

**Pull request closed without merging**